lib/elixir/lib/string.ex

```diff
@@ -674,15 +674,17 @@ defmodule String do
   end
 
   def upcase(string, :ascii) when is_binary(string) do
-    for <<x <- string>>,
-      do: if(x >= ?a and x <= ?z, do: <<x - 32>>, else: <<x>>),
-      into: ""
+    IO.iodata_to_binary(upcase_ascii(string))
   end
 
   def upcase(string, mode) when mode in @conditional_mappings do
     String.Casing.upcase(string, [], mode)
   end
 
```

Comment:
Im not with a computer but would `when char in ?a..?z, do:` work? If so that will read better

Reply:
@eksperimental normally, yes -- but as per @josevalim  using `in` at this level will cause bootstrapping issues.